src/Akeneo/Pim/Automation/DataQualityInsights/back/Infrastructure/Persistence/Query/ProductEnrichment/GetProductModelIdsFromProductModelCodesQuery.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Akeneo\Pim\Automation\DataQualityInsights\Infrastructure\Persistence\Query\ProductEnrichment;
+
+use Akeneo\Pim\Automation\DataQualityInsights\Domain\Query\ProductEnrichment\GetProductModelIdsFromProductModelCodesQueryInterface;
+use Akeneo\Pim\Automation\DataQualityInsights\Domain\ValueObject\ProductId;
+use Doctrine\DBAL\Connection;
+
+/**
+ * @copyright 2022 Akeneo SAS (https://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+final class GetProductModelIdsFromProductModelCodesQuery implements GetProductModelIdsFromProductModelCodesQueryInterface
+{
+    public function __construct(
+        private Connection $dbConnection,
+    ) {
+    }
+
+    public function execute(array $productModelCodes): array
```

Comment:
In order to cover the use of the query with static analysis of the code, could we define with more precision the return type by using PHPDoc?

```suggestion

    /**
     * @param string[] $productIdentifiers
     * @return array<string, ProductId>
     */
    public function execute(array $productModelCodes): array
```